infra/build/functions/sync/main_test.py

```diff
@@ -113,74 +157,165 @@ def setUpClass(cls):
         _DATASTORE_EMULATOR_PORT)
     os.environ['GOOGLE_CLOUD_PROJECT'] = _TEST_PROJECT_ID
     os.environ['DATASTORE_DATASET'] = _TEST_PROJECT_ID
+    os.environ['GCP_PROJECT'] = 'test-project'
+    os.environ['FUNCTION_REGION'] = 'us-central1'
 
-  def test_sync_projects(self):
-    """Testing sync_projects()."""
+  def test_sync_projects_update(self):
+    """Testing sync_projects() updating a schedule."""
     client = ndb.Client()
+    cloud_scheduler_client = CloudSchedulerClient()
 
     with client.context():
-      Project(name='test1').put()
-      Project(name='test2').put()
+      Project(name='test1', schedule='0 8 * * *').put()
+      Project(name='test2', schedule='0 9 * * *').put()
 
-      projects = {'test1', 'test3'}
-      sync_projects(projects)
+      projects = {
+          'test1': ProjectMetadata('0 8 * * *'),
+          'test2': ProjectMetadata('0 7 * * *')
+      }
+      sync_projects(cloud_scheduler_client, projects)
 
       projects_query = Project.query()
-      self.assertEqual(projects, {project.name for project in projects_query})
+      self.assertEqual({
+          'test1': '0 8 * * *',
+          'test2': '0 7 * * *'
+      }, {project.name: project.schedule for project in projects_query})
+      clean = [project.key for project in projects_query]
+      ndb.delete_multi(clean)
+
+  def test_sync_projects_create(self):
+    """"Testing sync_projects() creating new schedule."""
+    client = ndb.Client()
+    cloud_scheduler_client = CloudSchedulerClient()
+
+    with client.context():
+      Project(name='test1', schedule='0 8 * * *').put()
 
-    # Todo: Clean datastore in case of new tests relating to same entities.
+      projects = {
+          'test1': ProjectMetadata('0 8 * * *'),
+          'test2': ProjectMetadata('0 7 * * *')
+      }
+      sync_projects(cloud_scheduler_client, projects)
 
-  def test_get_projects(self):
-    """Testing get_projects()."""
+      projects_query = Project.query()
+      self.assertEqual({
+          'test1': '0 8 * * *',
+          'test2': '0 7 * * *'
+      }, {project.name: project.schedule for project in projects_query})
+      clean = [project.key for project in projects_query]
+      ndb.delete_multi(clean)
+
+  def test_sync_projects_delete(self):
+    """Testing sync_projects() deleting."""
+    client = ndb.Client()
+    cloud_scheduler_client = CloudSchedulerClient()
+
+    with client.context():
+      Project(name='test1', schedule='0 8 * * *').put()
+      Project(name='test2', schedule='0 9 * * *').put()
+
+      projects = {'test1': ProjectMetadata('0 8 * * *')}
+      sync_projects(cloud_scheduler_client, projects)
+
+      projects_query = Project.query()
+      self.assertEqual(
+          {'test1': '0 8 * * *'},
+          {project.name: project.schedule for project in projects_query})
+      clean = [project.key for project in projects_query]
```

Comment:
Rather than doing this let's fix up the emulator such that we reset it in setUp. Can do this in another PR. See https://github.com/google/clusterfuzz/blob/00e0263a3c62af1a8d5673d120388fbcb73e629f/src/python/tests/test_libs/test_utils.py#L176 for how to do this.